common/src/main/java/com/microsoft/identity/common/internal/configuration/LibraryConfiguration.java

```diff
@@ -0,0 +1,79 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.internal.configuration;
+
+import com.microsoft.identity.common.internal.logging.Logger;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.experimental.SuperBuilder;
+
+/**
+ * Holds configuration settings that are global in scope.
+ * Apply to all public client application instances and all requests.
+ */
+@Getter
+@EqualsAndHashCode()
+@SuperBuilder(toBuilder = true)
+public class LibraryConfiguration {
+
+    private static final String TAG = LibraryConfiguration.class.getSimpleName();
+    private static LibraryConfiguration sInstance = null;
+
+    // private constructor restricted to this class itself
+    private LibraryConfiguration() {
+
+    }
+
+    // static method to create instance of Singleton class
+    public synchronized static LibraryConfiguration getInstance() {
+        if (sInstance == null) {
+            sInstance = createDefaultInstance();
+        }
+
+        return sInstance;
+    }
+
+    public static synchronized void intializeLibraryConfiguration(LibraryConfiguration config){
+        if(sInstance == null){
+            sInstance = config;
+        }else{
+            Logger.warn(TAG, "MsalConfiguration was already initialized");
```

Comment:
should we throw a runtime exception?

What is the expected behavior if the user is trying to initialize with a different config? and effectively trying to change the config...seems like they are tied with what they selected in the first place.

Reply:
Yes.  This is a limitation of our configuration design.  We don't have a way to set global settings, but changing that now would be a breaking change.  I created a work item in the backlog for our Android 12 release to address this issue.  I'm fine with logging the warning and documenting this for now.

Reply:
Sounds good